src/javascript/JContent/EditFrame/Box.jsx

```diff
@@ -190,64 +192,70 @@ export const Box = React.memo(({
     const type = element.getAttribute('type');
 
     // Display current header through portal to be able to always position it on top of existing selection(s)
-    const headerProps = {
-        className: clsx(styles.headerContainer, isSticky && styles.sticky, 'flexRow_nowrap', 'alignCenter', editStyles.enablePointerEvents)
-    };
+    const headerStyles = clsx(
+        styles.boxHeader,
+        isSticky && styles.sticky,
+        'flexRow_nowrap',
+        'alignCenter',
+        editStyles.enablePointerEvents,
+        isClicked && styles.isClicked,
+        isHeaderHighlighted && styles.isHovered
+    );
 
     const Header = (
-        <div {...headerProps}
-             jahiatype="header" // eslint-disable-line react/no-unknown-property
-             data-current={isCurrent}
-             data-jahia-id={element.getAttribute('id')}
-             onMouseOver={onMouseOver}
-             onMouseOut={onMouseOut}
-             onClick={onClick}
-             onDoubleClick={onDoubleClick}
+        <header ref={dragWithChecks}
+                className={headerStyles}
+                jahiatype="header" // eslint-disable-line react/no-unknown-property
+                data-current={isCurrent}
+                data-clicked={isClicked}
+                data-hovered={isHeaderHighlighted}
```

Comment:
might be more appropriate to distinguish data-hovered (isCurrent) vs data-highlighted (isHeaderHighlighted)

Reply:
I renamed it `data-highlighted`